src/jit/codegenxarch.cpp

```diff
@@ -4386,9 +4386,11 @@ void CodeGen::genCodeForLclAddr(GenTree* tree)
     regNumber targetReg  = tree->gtRegNum;
 
     // Address of a local var.
-    noway_assert(targetType == TYP_BYREF);
+    noway_assert((targetType == TYP_BYREF) || (targetType == TYP_I_IMPL));
 
-    inst_RV_TT(INS_lea, targetReg, tree, 0, EA_BYREF);
+    emitAttr size = (targetType == TYP_BYREF) ? EA_BYREF : EA_PTRSIZE;
```

Comment:
emitTypeSize(targetType)

Reply:
Changed as suggested. I also added this change to codegenarmarch.